src/OrchardCore.Modules/OrchardCore.Contents/Handlers/FullTextAspectContentHandler.cs

```diff
@@ -62,12 +63,11 @@ public override Task GetContentItemAspectAsync(ContentItemAspectContext context)
 
                     if (bodyAspect != null && bodyAspect.Body != null)
                     {
-                        using (var sw = new StringWriter())
-                        {
-                            // Don't encode the body
-                            bodyAspect.Body.WriteTo(sw, NullHtmlEncoder.Default);
-                            fullTextAspect.Segments.Add(sw.ToString());
-                        }
+                        using var stringBuilderPool = StringBuilderPool.GetInstance();
+                        await using var sw = new StringWriter(stringBuilderPool.Builder);
```

Comment:
Does `StringWriter` implement `DisposeAsync()` ?

Reply:
It does through TextWriter, though it seems to be useless code inside around Dispose(), I think it's better to call Dispose().

Reply:
wouldn't compile otherwise, I will revert back to simple `using`

Reply:
> wouldn't compile otherwise

Oops ;)